### PR TITLE
VEOSS-598 Correction submission

### PR DESCRIPTION
**Pull request by DevAndyroo**

### CODE REVIEW

**Review by oliwiaKot — APPROVED** [review]

**Review by oliwiaKot — APPROVED** [review]

File: app/controllers/CorrectionController.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  def post(): Action[CorrectionRequest] = cc.authAndCorrectionToggle(parse.json[CorrectionRequest]).async {
+    implicit request =>
+      correctionService.createCorrection(request.body).map {
+        case Some(vatReturn) => Created(Json.toJson(vatReturn))
+        case None => Conflict
+      }
+  }
+
+  def list(): Action[AnyContent] = cc.authAndCorrectionToggle.async {
```
Comment (darbyo) [inline app/controllers/CorrectionController.scala:45]: Could make this just an overloaded get method on the controller the same as the service methods of the same name. Not fussed either way

File: app/repositories/CorrectionRepository.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    collection
+      .insertOne(correction)
+      .toFuture()
+      .map(_ => Some(correction))
+      .recover {
+        case Duplicate(_) => None
+      }
+  }
+
+  def get(vrn: Vrn): Future[Seq[CorrectionPayload]] =
```
Comment (darbyo) [inline app/repositories/CorrectionRepository.scala:60]: Can you add to a test that this method returns empty when no correction payloads exist and/or a test where correction exist in the db with a different VRN this should still return Seq.empty

File: app/repositories/CorrectionRepository.scala [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    collection
+      .find(Filters.equal("vrn", toBson(vrn)))
+      .toFuture()
+
+  def get(vrn: Vrn, period: Period): Future[Option[CorrectionPayload]] =
```
Comment (darbyo) [inline app/repositories/CorrectionRepository.scala:65]: Also add a test for this returning None

File: app/services/CorrectionService.scala [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      period = request.period,
+      corrections = request.corrections,
+      submissionReceived = Instant.now(clock),
+      lastUpdated = Instant.now(clock)
+    )
+
+    repository.insert(correctionPayload)
+  }
+
+  def get(vrn: Vrn): Future[Seq[CorrectionPayload]] =
```
Comment (darbyo) [inline app/services/CorrectionService.scala:47]: I know these are just proxy methods, but maybe add a couple super quick tests for regression. Aswell as on for createCorrection  returning None

**Review by darbyo — CHANGES_REQUESTED** [review]

File: app/controllers/CorrectionController.scala (hunk 1 above)
Comment (DevAndyroo) [reply app/controllers/CorrectionController.scala:45]: was keeping it in line with vat return controller (but I may have renamed that)
